src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_StringConcat.cs

```diff
@@ -398,7 +696,8 @@ private BoundExpression ConvertConcatExprToString(SyntaxNode syntax, BoundExpres
             }
 
             // If it's a string already, just return it
-            if (expr.Type.IsStringType())
+            // If it's a char, return it here, so we can apply span-based optimization later
+            if (expr.Type.IsStringType() || expr.Type.IsCharType())
```

Comment:
>|| expr.Type.IsCharType() [](http://example.com/codeflow?start=41&length=25)

It looks like we cannot benefit from leaving this expression as char unless we have ```WellKnownMember.System_ReadOnlySpan_T__ctor_Reference```. It might be much simpler to check for the helper right here. If we don't have it, we just keep going with the regular conversion. This way we don't need to worry about the lack of the helper later.

Reply:
For simplicity, it might make sense to check for ```WellKnownMember.System_String__Concat_ReadOnlySpanReadOnlySpan``` here as well. Most likely, even when we will be able to take advantage of Concat Span overloads with more parameters, we will allocate a string for the char, and then get span from it. To me, it doesn't feel worthwhile spending effort to support an optimization for scenarios like that.